ats/atsMachines/fluxScheduled.py

```diff
@@ -0,0 +1,101 @@
+#ATS:slurm56 SELF FluxScheduled 3
+#ATS:toss_4_x86_64_ib SELF FluxScheduled 16
+
+from ats.atsMachines import lcMachines
+import time
+from math import ceil
+
+## Must be on a Flux machine or with Flux broker running in Slurm allocation for these imports to work.
+
+import flux
+import flux.job
+f = flux.Flux() ## defines a persistent handle to use to connect to the broker
+
+## Do we need to request an allocation? On fluke if you issue Flux mini run it sends to a compute node automatically
+## but there's no assurances it will be to the _same_ compute nodes for each of ats's "tests" or whatever
+## answer: YES for now
+
+## Should start its own Flux session!! TODO
+
+class FluxScheduled(lcMachines.LCMachineCore):
+    def init(self):
+        ## anything to initialize here?
+        ## set ceiling on number of nodes and cores in the allocation.
+        self.maxNodes = int(flux.resource.list.resource_list(f).get().free.nnodes)
+        self.maxCores = int(flux.resource.list.resource_list(f).get().free.ncores)
+        self.coresPerNode = int(self.maxCores // self.maxNodes)
+
+        ## overwritten machine things I don't understand.
+        self.numberTestsRunningMax = self.maxCores
+        import pdb 
+        pdb.set_trace()
+        ## set number of nodes
+
+    def examineOptions(self, options):
+        ## Option values for Flux 
+        self.exclusive = options.exclusive
+
+    def calculateCommandList(self, test):
+        ## flux mini run commands will go here
+        ret = "flux mini run".split()
+        np = test.options.get('np', 1)
+        nn = test.options.get('nn', 1)
+        if np > self.coresPerNode:
+            nn = ceil(np / self.coresPerNode)
+        # import pdb 
+        # print("np trace")
+        # pdb.set_trace()
+        
+        # set number of nodes and cores
+        ret.append(f"-N {nn}")
+        ret.append(f"-n {np}")
```

Comment:
This works with Flux's command-line parsing, but you're combining the arguments in a way that some parsers might not like. I think this should either be `ret.append(f"-n{np}")` or `ret += ["-n", str(np)]`. When you see `-N 5` on the command line, that's really two arguments---that's all I'm getting at.

Reply:
Another option: `ret.extend(["-n", str(np)])`

Reply:
Thinking about this now that I'm adding thread support. In `flux mini run` this is specified as `--threads=1` for 1 thread, etc. With the `=` is this one argument, or two? How many is it supposed to be?

Reply:
I have never seen that `--threads` option, where are you getting that from? Threads aren't really a resource, so AFAIK Flux doesn't allocate them. User's applications are free to create however many threads they want.

Reply:
When I said that `-N 5` is really two arguments, what I meant was that if you execute `./foo -N 5` in your shell, `argc` for that process will be 3. `argv[1]` would be `-N` and `argv[2]` would be `-5`.

If you want to play around with this you could try:

```bash
echo "import sys; print(sys.argv)" > script.py
bash-5.1$ python3 script.py --threads=1
['script.py', '--threads=1']
```

so as you can see `--threads=1` is one argument